test/files/run/package-object-toolbox.scala

```diff
@@ -0,0 +1,40 @@
+import java.io.File
+import java.net.URLClassLoader
+
+import scala.reflect.io.Path
+import scala.reflect.runtime.{ universe => ru }
+import scala.tools.partest._
+import scala.tools.reflect.ToolBox
+
+import org.junit.Assert._
+
+object Test extends StoreReporterDirectTest {
+  val cp = List(sys.props("partest.lib"), testOutput.path)
+  override def extraSettings = s"-cp ${cp.mkString(File.pathSeparator)}"
+
+  def show(): Unit = {
+    compiles("package object pkg { def foo = 1 }")
+    val loader = new URLClassLoader(cp.map(new File(_).toURI.toURL).toArray)
+    val mirror = ru.runtimeMirror(loader)
+
+    val toolbox = mirror.mkToolBox()
+    val result1 = toolbox.eval(toolbox.parse("pkg.foo"))
+    assertEquals(1, result1)
+
+    val obj  = toolbox.eval(toolbox.parse("pkg.`package`"))
+    val pkg  = mirror.staticPackage("pkg")
+    val sym  = pkg.info.decl(ru.TermName("foo")).asMethod
+    val meth = mirror.reflect(obj).reflectMethod(sym)
+    val res2 = meth.apply()
+    assertEquals(1, res2)
```

Comment:
@retronym I tried this, but then _not_ overriding in ReflectGlobal doesn't fail the test, which isn't a good sign...

Reply:
I was worried about `Toolbox.typecheck` which has a fairly bespoke way of using the compiler. But adding that variation to this test works fine, with or without the override in `ReflectGlobal`. Without the override the condition `isPast(currentRun.namerPhase)` holds.

That comes done to:

```scala
          phase = run.typerPhase // need to set a phase to something <= typerPhase, otherwise implicits in typedSelect will be disabled
          globalPhase = run.typerPhase // amazing... looks like phase and globalPhase are different things, so we need to set them separately

```

There isn't an API in the toolbox (currently) that allows you to define a top-level package, everything gets wrapped in a synthetic package. So I don't think there is a way to trigger the original bug in toolbox anyway. The test here is good to make sure that package objects are working in toolbox in any case.

Should we remove the override given that it isn't strictly necessary? I guess so.